assignments/lambda-classes.js

```diff
@@ -1 +1,182 @@
 // CODE here for your Lambda Classes
+
+class Person {
+    constructor(generalInfo) {
+        this.name = generalInfo.name;
+        this.age = generalInfo.age;
+        this.location = generalInfo.location;
+        this.gender = generalInfo.gender;
+    }
+
+    speak() {
+        console.log(`Hello my name is ${this.name} and I am from ${this.location}.`); 
+    }
+}
+
+class Instructor extends Person {
+    constructor(instructorInfo) {
+        super(instructorInfo);
+        this.specialty = instructorInfo.specialty;
+        this.favLanguage = instructorInfo.favLanguage;
+        this.catchPhrase = instructorInfo.catchPhrase;
+    }
+    demo(subject) {
+        console.log(`Today we are learning about ${subject}.`);
+    }
+    grade(studentObj, subject) {
+        console.log(`${studentObj.name} receives a perfect score on ${subject}.`);
+    }
+}
+
+class Student extends Person {
+    constructor(studentInfo) {
+        super(studentInfo);
+        this.previousbackground = studentInfo.previousbackground;
+        this.className = studentInfo.className;
+        this.favSubjects = studentInfo.favSubjects;
+    }
+
+    listsSubjects() {
+        this.favSubjects.forEach(subject => {
+            console.log(subject);
```

Comment:
You can format this console into a sentence structure for better context readability.